### DISCUSSION TITLE
Is it possible to hook into the compiler to customize module resolution?

### DISCUSSION THREAD

**Discussion by helmturner** [tamagui/tamagui#2127] [Q&A]
I'd prefer to use the `imports` field in package.json over tsconfig paths. I have all of my bundlers successfully resolving them, but the Tamagui compiler chokes. Is there anything I can do about that?

**Comment by natew** [comment]
This is actually something I've been wanting to fix. What would be ideal, right now the Tamagui compiler uses esbuild, does it not detect your tsconfig? Do you want hooks into the esbuild config?

**Reply by helmturner** [reply]
It does pick up my TSConfig, as I'm able to 'simulate' `package.json#imports` with an alias (`#ui` etc).

Ultimately, that's what I'm going for. Next/webpack and typescript (if configured correctly) all resolve the imports field, and it's trivial to add the functionality to Expo's `Metro`.

Unfortunately, such a configuration breaks the Tamagui compiler (package resolution fails).

If ESbuild is supposed to respect the `resolvePackageImports` field in my tsconfig automatically, I guess it would be an ESbuild issue... But as far as I know it requires adding a [plug-in to hook into ESBuild's resolver](https://esbuild.github.io/plugins/#resolve).

**Comment by helmturner** [comment]
Just dug  bit deeper into the docs on [tsup](https://github.com/egoist/tsup), and it seems like a nice extension of tsbuild that covers some edge-cases and limitations. I'm wondering what the effort/payoff ratio would look like for migrating... I might be able to start playing around with it a bit.
